### DISCUSSION TITLE
Node 16.18.0 upgrade needed for Homebridge UI

### DISCUSSION THREAD

**Discussion by databoy01** [dgreif/ring#1298] [Q&A] [answered]
So my Homebridge UI wants to do an upgrade and it tells me to upgrade my Node. Now in the past upgrading the node could break the Ring Plugin, so I am apprehensive in upgrading that…. Anyone upgrade to the newest node and still have no issues with Ring?

This is what the upgrade says btw: 

> “The homebridge-config-ui-x plugin recommends running on Node.js v16.20.2 or later. You are currently running Node.js v16.18.0.                                                             Are you sure you want to update `homebridge-config-ui-x?”

Of course I haven’t done it yet as an fyi.

**Comment by tsightler** [comment]
Well, I wouldn't upgrade to the "newest" node, which is like node v20.5.1, but you should generally be safe upgrading to the lastest versions of existing maintenance or LTS releases.  Note that node v16.x is soon (less than a month) to be out of support while the current LTS version is v18.17.1.  I'm running v18.17.1 on my Homebridge with no issues and v18/20 are the recommended versions for homebridge-ring.  I always recommend being on the most recent LTS release as a baseline.

**Reply by dgreif** [reply, marked as answer]
Yep, the v12 beta drops official Node 16 support and requires at least Node 18. I would also recommend staying on the latest LTS version

**Comment by databoy01** [comment]
Thanks @tsightler and @dgreif, updated to 18.17.1 no problem everything works still. As they say, I’d rather be sure than a find out the hard way…. Thanks again.
